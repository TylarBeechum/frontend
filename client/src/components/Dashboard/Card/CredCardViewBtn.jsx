import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { BaseButton } from '../../../styles/themes';

const CredCardViewBtn = ({ getModal }) => {
  return (
    <CardCredBtnContainer>
      <CardCredViewButton
        label="View"
        a11yTitle="view credentials button"
        onClick={getModal}
      />
    </CardCredBtnContainer>
  );
};

CredCardViewBtn.propTypes = {
  getModal: PropTypes.func.isRequired
};

const CardCredBtnContainer = styled.div`
  display: flex;
  align-items: center;
  margin-left: 3%;
  width: 15%;
`;

const CardCredViewButton = styled(BaseButton)`
  border-radius: 50px;
  border-color: #adadad;

  :hover {
    border-color: #ad91ed;
    background-color: #ad91ed;
    color: white;
  }
`;

export default CredCardViewBtn;
