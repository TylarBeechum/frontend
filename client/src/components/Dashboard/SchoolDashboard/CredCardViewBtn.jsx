import React from 'react';
import styled from 'styled-components';
import { BaseButton } from '../../../styles/themes';

const CredCardViewBtn = () => {
  return (
    <CardCredBtnContainer>
      <CardCredViewButton label="View" a11yTitle="view credentials button" />
    </CardCredBtnContainer>
  );
};

const CardCredBtnContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  margin-left: 3%;
`;

const CardCredViewButton = styled(BaseButton)`
  border-radius: 5%;
  border-color: #e3e3e3;

  :hover {
    border-color: #7d4cdb;
    background-color: #7d4cdb;
    color: white;
  }
`;

export default CredCardViewBtn;
