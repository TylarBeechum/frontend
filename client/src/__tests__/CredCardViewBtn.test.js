/* eslint-disable */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { fireEvent } from '@testing-library/react';
import {
  renderWithRouterAndProviders,
  rendererWithRouterAndProviders
} from '../testHelpers';

import CredCardViewBtn from '../components/Dashboard/Card/CredCardViewBtn';

describe('<CredCardViewBtn />', () => {
  it('should render the component to the screen successfully without crashing', () => {
    const spy = jest.fn();
    const helpers = renderWithRouterAndProviders(
      <CredCardViewBtn getModal={spy} />
    );
  });

  it('should match the previous snapshot of the component', () => {
    const mock = jest.fn();
    const tree = rendererWithRouterAndProviders(
      <CredCardViewBtn getModal={mock} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should render the View string inside the button to the screen', () => {
    const mock = jest.fn();
    const { getByText } = renderWithRouterAndProviders(
      <CredCardViewBtn getModal={mock} />
    );
    expect(getByText(/View/i)).toBeDefined();
  });

  it('should fire the function when the button is clicked', () => {
    const mock = jest.fn();
    const { getByText } = renderWithRouterAndProviders(
      <CredCardViewBtn getModal={mock} />
    );
    const viewBtn = getByText(/View/i);
    fireEvent.click(viewBtn);
    expect(mock).toHaveBeenCalledTimes(1);
  });
});
