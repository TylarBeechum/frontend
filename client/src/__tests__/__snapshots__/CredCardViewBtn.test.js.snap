// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CredCardViewBtn /> should match the previous snapshot of the component tests that the component has not changed and matched the previous snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 iDwiuh"
>
  <div
    className="sc-EHOje hWneJQ"
  >
    <button
      aria-label="view credentials button"
      className="sc-bdVaJa sc-bZQynM loBWCV StyledButton-sc-323bzc-0 kHrfdt"
      onBlur={[Function]}
      onClick={[MockFunction]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      type="button"
    >
      View
    </button>
  </div>
</div>
`;
